src/java.base/share/classes/com/sun/crypto/provider/TlsKeyMaterialGenerator.java

```diff
@@ -86,16 +87,18 @@ protected SecretKey engineGenerateKey() {
             throw new IllegalStateException(
                 "TlsKeyMaterialGenerator must be initialized");
         }
+        byte[] masterSecret = spec.getMasterSecret().getEncoded();
         try {
-            return engineGenerateKey0();
+            return engineGenerateKey0(masterSecret);
         } catch (GeneralSecurityException e) {
             throw new ProviderException(e);
+        } finally {
+            Arrays.fill(masterSecret, (byte)0);
         }
```

Comment:
It seems that for other Tls* classes, the Arrays.fill(...) call is still inside each method instead of being moved up a level. Just curious why this is done differently?

Reply:
The `engineGenerateKey0` method is quite long and I don't want to wrap everything in a big try-finally block, so I move it a little higher. Now `masterSecret` is still created and cleaned in the same method.

Reply:
Ok